crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -78,76 +74,81 @@ def repositories(platform)
       def admin_ip
         NodeObject.admin_node.ip
       end
+
+      def web_port
+        Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
+      end
     end
 
     def initialize(platform, repo)
       @platform = platform
       @id = repo
-      @registry = Repository.registry[@platform]["repos"][@id]
+      @config = Repository.registry[@platform]["repos"][@id]
       @url = url
     end
 
-    def available?
-      all_repo_dirs.include?(@registry["name"]) && check_repo_tag
+    def is_remote?
+      !@config["url"].blank?
     end
 
-    def valid_key_file?
-      if File.exist?(repomd_key_path)
-        md5 = Digest::MD5.hexdigest(File.read(repomd_key_path))
-        repomd_key_md5 == md5
-      else
-        false
-      end
+    def available?
+      is_remote? || (check_directory && check_repo_tag && check_key_file)
     end
 
-    def repodata_path
-      "/srv/tftpboot/#{@platform}/repos/#{@registry['name']}/repodata"
+    def exists?
+      is_remote? || check_directory
     end
 
-    def repomd_key_md5
-      @registry["repomd"]["md5"]
+    def valid_repo?
+      is_remote? || check_repo_tag
     end
 
-    def repomd_key_path
-      @registry["repomd"]["key"] || "#{repodata_path}/repomd.xml.key"
+    def valid_key_file?
+      is_remote? || check_key_file
     end
 
     def url
-      @registry["url"] || \
-        "http://#{Repository.admin_ip}:#{Repository.web_port}/#{@platform}/repos/#{@registry['name']}"
+      @config["url"] || \
+        "http://#{Repository.admin_ip}:#{Repository.web_port}/#{@platform}/repos/#{@config['name']}"
     end
 
     def active?
-      !bag_item.nil?
-    end
+      active_repos = Chef::DataBag.load("crowbar/repositories") rescue {}
```

Comment:
Style/RescueModifier: Avoid using `rescue` in its modifier form. (https://github.com/bbatsov/ruby-style-guide#no-rescue-modifiers)